src/api/Buffer.test.ts

```diff
@@ -28,6 +28,25 @@ describe('Buffer API', () => {
   let proc;
   let nvim: NeovimClient;
 
+  // utility to allow each test to be run in its 
+  // own buffer
+  function withBuffer(lines,test) { 
+    return async () => {
+        await nvim.command('new');
```

Comment:
Aaaah, but that's where the asynchronous part can stab us. 

At the beginning of the test, when we call `withBuffer`, the current buffer is either the original buffer or the buffer of the previous test -- which because it's all asynchronous, we can't say for sure if it's done or not. If it's not done and we do `enew!`, it won't be happy. 

Which is why `withBuffer` tries to leave all previous open buffers untouched. It is meant to create its own buffer and clean after itself, doing its best not to create side-effects.